pandas/core/internals/__init__.py

```diff
@@ -45,6 +47,10 @@ def __getattr__(name: str):
             # on hard-coding stacklevel
             stacklevel=2,
         )
+        if name == "DatetimeTZBlock":
+            from pandas.core.internals.api import _DatetimeTZBlock as DatetimeTZBlock
+
+            return DatetimeTZBlock
```

Comment:
This deprecated import was removed in https://github.com/pandas-dev/pandas/pull/58467, which was temporarily reverted in https://github.com/pandas-dev/pandas/pull/58715 to keep compatibility with our oldest supported pyarrow versions (this is no longer needed starting from pyarrow 15). 
But when reverting, we only reverted part of the original PR: adding back ExtensionBlock, and not DatetimeTZBlock, while it was actually DatetimeTZBlock which pyarrow needed (the use of ExtensionBlock was already removed in a much older version of pyarrow). Not sure how I missed that in the original revert (we were also skipping the parquet tests that needed this for the then oldest supported pyarrow version .. that skip is now removed)